File: app/src/main/java/reschikov/geekbrains/androidAdvancedLevel/weatherApplication/data/apiux/Forecast.java
package reschikov.geekbrains.androidadvancedlevel.weatherapplication.data.apiux;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Forecast implements Parcelable {

    @SerializedName("forecastday")
    @Expose
    private final List<Forecastday> forecastday;

    public Forecast(List<Forecastday> forecastday) {
        this.forecastday = forecastday;
    }

    private Forecast(Parcel in) {
        forecastday = in.createTypedArrayList(Forecastday.CREATOR);
    }

    public static final Creator<Forecast> CREATOR = new Creator<Forecast>() {
        @Override
        public Forecast createFromParcel(Parcel in) {
            return new Forecast(in);
        }

        @Override
        public Forecast[] newArray(int size) {
            return new Forecast[size];
        }
    };

    public List<Forecastday> getForecastday() {
        return forecastday;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeTypedList(forecastday);
    }
}

File: app/src/main/java/reschikov/geekbrains/androidadvancedlevel/weatherapplication/data/network/model/data/openweather/forecast/Forecast.kt
package reschikov.geekbrains.androidadvancedlevel.weatherapplication.data.network.model.data.openweather.forecast

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import reschikov.geekbrains.androidadvancedlevel.weatherapplication.data.network.model.data.openweather.current.*

class Forecast(@SerializedName("dt") @Expose val dt: Int,
               @SerializedName("main") @Expose val main: Main,
               @SerializedName("weather") @Expose val weather: List<Weather>,
               @SerializedName("clouds") @Expose val clouds: Clouds?,
               @SerializedName("wind") @Expose val wind: Wind,
               @SerializedName("rain") @Expose val rain: Rain?,
               @SerializedName("snow") @Expose val snow: Snow?,
               @SerializedName("dt_txt") @Expose val dt_txt: String)